### PR TITLE
feat(ps-card): added the ps-card and the ps-header

### PR DESCRIPTION
**Pull request by 6rayWolf**

### CODE REVIEW

File: projects/components/card/src/card.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    expect(await header.getCaptionText()).toBe('foo');
+  });
+
+  it('should show caption template', async () => {
+    expect(await header.getCaptionText()).toBeFalsy();
+
+    component.addCaptionTemplate = true;
+    const nodes = await header.getCaptionTemplateNodes();
+    expect(nodes.length).toEqual(1);
+    expect(await nodes[0].text()).toEqual('customCaption');
```
Comment (saithis) [inline projects/components/card/src/card.component.spec.ts:80]: maybe you could check, if it is a h1 node

File: projects/components/card/src/card.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  });
+
+  it('should show top buttons', async () => {
+    let nodes = await header.getActionTemplateNodes();
+    expect(nodes.length).toBe(0);
+
+    component.addTopButton = true;
+    nodes = await header.getActionTemplateNodes();
+    expect(nodes.length).toBe(1);
+    expect(await nodes[0].text()).toEqual('testTopButton');
```
Comment (saithis) [inline projects/components/card/src/card.component.spec.ts:128]: maybe check if it is a span node

File: projects/components/card/src/card.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    let nodes = await card.getActionTemplateNodes();
+    expect(nodes.length).toBe(0);
+
+    component.addCardActionButton = true;
+    nodes = await card.getActionTemplateNodes();
+    expect(nodes.length).toBe(1);
+    expect(await nodes[0].text()).toEqual('testActionButton');
+  });
+
+  it('should show action buttons', async () => {
```
Comment (saithis) [inline projects/components/card/src/card.component.spec.ts:141]: wrong text^^

File: projects/components/card/src/testing/card.harness.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  static with(options: PsCardHarnessFilters = {}): HarnessPredicate<PsCardHarness> {
+    return new HarnessPredicate(PsCardHarness, options)
+      .addOption('caption', options.caption, (harness, title) => HarnessPredicate.stringMatches(harness.getCaptionText(), title))
+      .addOption('description', options.description, (harness, subtitle) =>
+        HarnessPredicate.stringMatches(harness.getDescriptionText(), subtitle)
+      );
+  }
+
+  async getCaptionText(): Promise<string> {
```
Comment (saithis) [inline projects/components/card/src/testing/card.harness.ts:29]: maybe only a getHeader method, then you dont have to forward all header harness methods

File: projects/components/header/src/header.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    expect(await header.getCaptionText()).toBe('foo');
+  });
+
+  it('should show caption template', async () => {
+    expect(await header.getCaptionText()).toBeFalsy();
+
+    component.addCaptionTemplate = true;
+    const nodes = await header.getCaptionTemplateNodes();
+    expect(nodes.length).toEqual(1);
+    expect(await nodes[0].text()).toEqual('caption text');
```
Comment (saithis) [inline projects/components/header/src/header.component.spec.ts:67]: also check for h1?

File: projects/components/header/src/header.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    expect(await header.getDescriptionText()).toBe('bar');
+  });
+
+  it('should show description template', async () => {
+    expect(await header.getDescriptionText()).toBeFalsy();
+
+    component.addDescriptionTemplate = true;
+    const nodes = await header.getDescriptionTemplateNodes();
+    expect(nodes.length).toEqual(1);
+    expect(await nodes[0].text()).toEqual('description text');
```
Comment (saithis) [inline projects/components/header/src/header.component.spec.ts:94]: also check for span?

File: projects/components/header/src/header.directives.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { Directive } from '@angular/core';
+
+@Directive({
+  selector: '[psHeaderTopButtonSection]',
```
Comment (saithis) [inline projects/components/header/src/header.directives.ts:4]: maybe we should be more careful when naming template directives. in ps-table we use a mix of Template, Section and (nothing) as suffix. (*psTableColumnHeaderTemplate, *psTableTopButtonSection, *psTableListActions, etc.)

**Review by saithis — CHANGES_REQUESTED** [review]

**Review by Yirica — COMMENTED** [review]
Nothing else found, please fix the comments from saithis

File: projects/components/header/src/header.directives.ts (hunk 7 above)
Comment (6rayWolf) [reply projects/components/header/src/header.directives.ts:4]: We should talk about a convention for directive names and fix them all at once.

**Review by saithis — APPROVED** [review]

**Pull request merged**